[ADVAPP-2]: Enforce License Rules for SMS

Orrison opened the pull request:
### Ticket(s) or GitHub Issue

- https://canyongbs.atlassian.net/browse/ADVAPP-2

### Technical Description

Enforce License rules for SMS Notifications.

### Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Content or styling update (Changes which don't affect functionality)
- [ ] DevOps
- [ ] Documentation

### Screenshots (if appropriate)

### Any deployment steps required?

A deployment step could be a command that needs to be executed or an ENV key that needs to be added, for example.

- [ ] Yes, please specify
- [x] No

_______________________________________________

## Before contributing and submitting this PR, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/canyongbs/advisingapp/blob/main/README.md#contributing).
* [x] Title the PR with the ticket/issue number and a short description of the changes made. Or if no ticket/issue exists, title the PR with a short description of the changes made
* [x] Linked a relevant ticket or issue or describe the issue/feature which this PR resolves/implements.
* [x] Resolved all conflicts, if any.
* [x] Rebased your branch PR on top of the latest upstream `develop` branch.

dgoetzit commented on app-modules/notification/src/Notifications/Channels/SmsChannel.php:
```diff
@@ -114,4 +134,19 @@ public static function afterSending(object $notifiable, OutboundDeliverable $del
+
+        $resetWindow = $licenseSettings->data->limits->getResetWindow();
+
+        $currentQuoteUsage = OutboundDeliverable::where('channel', NotificationChannel::Sms)
```
```suggestion
        $currentQuotaUsage = OutboundDeliverable::where('channel', NotificationChannel::Sms)
```

dgoetzit reviewed (approved):
This looks great! Just a small variable change

Orrison commented:
@dgoetzit I have resolved the variable feedback and also I fixed an issue where we would end up forever trying to send an engagement that was `rate_limited`.

I also removed the check for whether or not the before hook returns false for now, since I don't believe there was currently a scenario where it would return false.

We can bring it back, or something similar later if we need to stop the email sending at that point.
